internal/aws/metrics/metric_calculator.go

```diff
@@ -35,12 +35,14 @@ func NewFloat64DeltaCalculator() MetricCalculator {
 }
 
 func calculateDelta(prev *MetricValue, val interface{}, timestamp time.Time) (interface{}, bool) {
-	deltaValue := val.(float64)
+	var deltaValue float64
 	if prev != nil {
-		deltaValue = deltaValue - prev.RawValue.(float64)
+		deltaValue = val.(float64) - prev.RawValue.(float64)
 		if deltaValue < 0 {
-			return float64(0), true
+			deltaValue = float64(0)
```

Comment:
Good point!
I think as a delta calculator, it should not represent the invoker, and decide whether a delta value needs to be reset. The case you mentioned never happened and I wouldn't expect it to happen. With said, it would be a reasonable guess if the condition `deltaValue < 0` gets hit. I moved this part of logic to `func (dps (Double|Int)DataPointSlice) At(i int) DataPoint`. Thanks!